crates/rspack_core/src/exports_info.rs

```diff
@@ -1829,3 +1829,247 @@ macro_rules! debug_exports_info {
     }
   };
 }
+
+pub trait ModuleGraphAccessor<'a> {
+  fn run_resolve_filter(
+    &self,
+    target: &ResolvedExportInfoTarget,
+    resolve_filter: &ResolveFilterFnTy,
+  ) -> bool;
+  fn get_export_info(&self, export_info_id: &ExportInfoId) -> Arc<ExportInfo>;
+  fn get_export_info_id(
+    &self,
+    name: &Atom,
+    module_identifier: &ModuleIdentifier,
+  ) -> Arc<ExportInfoId>;
+  fn get_max_target(
+    &self,
+    export_info_id: &ExportInfoId,
+  ) -> Arc<HashMap<Option<DependencyId>, ExportInfoTargetValue>>;
+  fn get_module_meta_exports_type(
+    &self,
+    module_identifier: &ModuleIdentifier,
+  ) -> Option<Arc<BuildMetaExportsType>>;
+  fn get_read_only_export_info(
+    &self,
+    name: &Atom,
+    module_identifier: &ModuleIdentifier,
+  ) -> Option<Arc<ExportInfo>>;
+}
+
+pub struct MutableModuleGraph<'a> {
+  inner: Mutex<&'a mut ModuleGraph>,
+}
+
+impl<'a> MutableModuleGraph<'a> {
+  pub fn new(mg: &'a mut ModuleGraph) -> Self {
+    Self {
+      inner: Mutex::new(mg),
+    }
+  }
+}
+
+impl<'a> ModuleGraphAccessor<'a> for MutableModuleGraph<'a> {
+  fn get_export_info_id(
```

Comment:
acquiring a lock every time for export info-related operations will introduce none-trivial regression.

Reply:
The exports_info is modified in get_export_info_id method. Is there any recommended way?

Reply:
Perhaps I should lock once and get the mutable module graph reference and unlock after all get_export_info_id calls?

Reply:
Added MutexModuleGraph to generate ModuleGraphAccessor after locking only once